examples/serialization/portable/main.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+const factoryID = 1
+const employeeClassID = 1
+
+type EmployeePortable struct {
+	Name string
+	Age  int32
+}
+
+func (e EmployeePortable) FactoryID() int32 {
+	return factoryID
+}
+
+func (e EmployeePortable) ClassID() int32 {
+	return employeeClassID
+}
+
+func (e EmployeePortable) WritePortable(writer serialization.PortableWriter) {
+	writer.WriteString("name", e.Name)
+	writer.WriteInt32("age", e.Age)
+}
+
+func (e *EmployeePortable) ReadPortable(reader serialization.PortableReader) {
```

Comment:
Pointer receiver allows the the function to change the struct they are attached to. Since `ReadPortable` mutates the fields of `EmployeePortable` it should take a pointer receiver.

Reply:
Forgive my Golang ignorance but I tried the following example https://play.golang.org/p/ZKLaWi7mf3u
Basically, I defined an interface with two methods and a struct. I have implemented one interface method with a pointer receiver and one with a struct receiver. The interface check says that my struct does not implement the interface.

So, my question is, if we do that, does the `EmployeePortable` passes this check https://github.com/hazelcast/hazelcast-go-client/blob/18e5440f55cdc69832bdaa856acfd4008e4d4278/internal/serialization/serialization.go#L285 ?

Reply:
In your example, `baz` takes a pointer, so the type assertation requires a pointer-type value. If you change your code to use `&F{}` instead of `F{}` then it will print true: https://play.golang.org/p/JUyNLkyvqqR

In the sample code, the pointer to the actual value is passed to `Set` so there's no problem:
```
employee := &EmployeePortable{Name: "Jane Doe", Age: 30}
```

Reply:
Oh, I see. Thanks for the clarification